editor/plugins/texture_editor_plugin.cpp

```diff
@@ -61,11 +70,27 @@ void TexturePreview::_notification(int p_what) {
 				metadata_label->add_theme_font_override(SceneStringName(font), metadata_label_font);
 			}
 
+			bg_rect->set_color(get_theme_color(SNAME("dark_color_2"), EditorStringName(Editor)));
 			checkerboard->set_texture(get_editor_theme_icon(SNAME("Checkerboard")));
+			cached_outline_color = get_theme_color(SNAME("extra_border_color_1"), EditorStringName(Editor));
 		} break;
 	}
 }
 
+void TexturePreview::_draw_outline() {
+	if (outline_width == 0) {
+		return;
+	}
+	const Rect2 outline_rect = Rect2(Vector2(), texture_display->get_size()).grow((float)outline_width / 2);
+	texture_display->draw_rect(outline_rect, cached_outline_color, false, (float)outline_width);
```

Comment:
Scale the outline width on hiDPI displays:

```suggestion
	const float scaled_width = Math::round(outline_width * EDSCALE);
	const Rect2 outline_rect = Rect2(Vector2(), texture_display->get_size()).grow(scaled_width * 0.5);
	texture_display->draw_rect(outline_rect, cached_outline_color, false, scaled_width);
```

Reply:
@Calinou I tested it on tablet (11', 2560x1600) and phone (6.53', 2340x1080) and it looked fine without taking EDSCALE into account.
If respecting EDSCALE is preferable, could you please tell – am I right that EDSCALE could not be changed at runtime (without editor restart)? Because this "outline" also affects `MarginContainer` margins.

Reply:
`EDSCALE` shouldn't change at runtime, the setting for editor scale requires restarting the editor to be taken into account.

Reply:
@akien-mga got it, thanks! I thought as much.
Will apply suggestions ASAP.